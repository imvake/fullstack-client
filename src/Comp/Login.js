import React, { useEffect, useState } from "react";
import {
  Button,
  Col,
  Container,
  Form,
  FormGroup,
  Input,
  Label,
  Row,
} from "reactstrap";
import logo from "../images/logo.png";
import { LoginValidations } from "../Validations/LoginValidations";
import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import { Link, useNavigate } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { getUser } from "../Features/UserSlice";
function Login() {
  let [email, setEmail] = useState("");
  let [password, setPassword] = useState("");
  let Navigate = useNavigate();
  const dispatch = useDispatch();
  const {
    register,
    handleSubmit: submitForm,
    formState: { errors },
  } = useForm({ resolver: yupResolver(LoginValidations) });

  const message = useSelector((state) => state.user.message);
  const userData = useSelector((state) => state.user.users);
  const isSuccess = useSelector((state) => state.user.isSuccess);
  const isError = useSelector((state) => state.user.isError);

  const handleSubmit = (e) => {
    const user = { email: email, password: password };
    dispatch(getUser(user));
  };

  useEffect(() => {
    if (userData && isSuccess) {
      Navigate("/Home");
    }
    if (isError) {
      Navigate("/");
    }
  }, [isSuccess, isError, userData]);
  return (
    <Container fluid>
      <Row className="div-row">
        <Col className="div-col" md="6">
          <form className="div-form">
            <div className="div-logo">
              <img
                src={logo}
                alt="Logo"
                className="logo"
                width="200px"
                height="100px"
              />
            </div>
            <FormGroup>
              <Label for="email">Email</Label>
              <input
                id="email"
                type="email"
                className="form-control"
                {...register("email", {
                  onChange: (e) => setEmail(e.target.value),
                })}
                value={email}
              />

              {errors.email && (
                <p className="alert alert-danger">{errors.email?.message}</p>
              )}
            </FormGroup>
            <FormGroup>
              <Label for="password">Password</Label>
              <input
                id="password"
                type="password"
                className="form-control"
                {...register("password", {
                  onChange: (e) => setPassword(e.target.value),
                })}
                value={password}
              />
              {errors.password && (
                <p className="alert alert-danger">{errors.password?.message}</p>
              )}
            </FormGroup>
            <FormGroup>
              <Input type="checkbox" />
              <Label>Remember me</Label>
            </FormGroup>
            <FormGroup>
              <Label>
                <a href="">Forget Password ?</a>
              </Label>
            </FormGroup>
            <FormGroup>
              <Button color="primary" onClick={submitForm(handleSubmit)}>
                SignIn
              </Button>
            </FormGroup>
            <FormGroup>
              <Label>
                <Link to="/Registration">No Account? SignUp</Link>
              </Label>
            </FormGroup>
          </form>
        </Col>
      </Row>
    </Container>
  );
}
export default Login;
