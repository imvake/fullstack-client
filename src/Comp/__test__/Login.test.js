import renderer from "react-test-renderer";
import Login from "../Login";
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { store } from "../../store";
import reducer from "../../Features/UserSlice";

/*test("Match the UI",()=>{
    const {container}=render(
        <Provider store={store}>
            <Router>
                <Login/>
            </Router>
        </Provider>
    );
    screen.debug(container);
    expect(container).toMatchSnapshot();
});*/

// test("Checking Email Input", () => {
//   render(
//     <Provider store={store}>
//       <Router>
//         <Login />
//       </Router>
//     </Provider>
//   );
//   const emailInput = screen.getByLabelText(/email/i);
//   //const testemail="abc@gmail.com";
//   //expect(emailInput.value).toEqual(testemail);
//   const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
//   fireEvent.change(emailInput, {
//     target: {
//       value: "abc@gmail.com",
//     },
//   });
//   expect(emailRegex.test(emailInput.value)).toBe(true);
// });

// test("Checking Password Input", () => {
//   render(
//     <Provider store={store}>
//       <Router>
//         <Login />
//       </Router>
//     </Provider>
//   );
//   const passInput = screen.getByLabelText(/password/i);
//   //const testemail="abc@gmail.com";
//   //expect(emailInput.value).toEqual(testemail);
//   const passwordRegex =
//     /^(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*(),.?":{}|<>]).{6,}$/;
//   fireEvent.change(passInput, {
//     target: {
//       value: "Abc@123",
//     },
//   });
//   expect(passwordRegex.test(passInput.value)).toBe(true);
// });

const testInitValue = {
  users: {},
  message: "",
  isLoading: false,
  isSuccess: false,
  isError: false,
};

test("Cechking the states", () => {
  expect(reducer(undefined, { type: undefined })).toEqual(testInitValue);
});
