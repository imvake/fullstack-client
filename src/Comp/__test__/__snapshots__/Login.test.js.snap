// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the UI 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="div-row row"
    >
      <div
        class="div-col col-md-6"
      >
        <form
          class="div-form"
        >
          <div
            class="div-logo"
          >
            <img
              alt="Logo"
              class="logo"
              height="100px"
              src="logo.png"
              width="200px"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
              id="email"
            >
              Email
            </label>
            <input
              class="form-control"
              name="email"
              type="email"
              value="abc213@gmail.com"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
              id="password"
            >
              Password
            </label>
            <input
              class="form-control"
              name="password"
              type="password"
            />
          </div>
          <div
            class="mb-3"
          >
            <input
              class="form-check-input"
              type="checkbox"
            />
            <label
              class="form-label"
            >
              Remember me
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              <a
                href=""
              >
                Forget Password ?
              </a>
            </label>
          </div>
          <div
            class="mb-3"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              SignIn
            </button>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              <a
                href="/Registration"
              >
                No Account? SignUp
              </a>
            </label>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
